File: app/src/main/java/android/montreal/devfest/stuff/policies/PoliciesModule.java
package android.montreal.devfest.stuff.policies;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class PoliciesModule {
    @Provides
    static RandomStuffVerifierImpl randomStuffVerifierImpl() {
        return new RandomStuffVerifierImpl();
    }

    @Binds
    abstract RandomStuffVerifier randomStuffVerifier(RandomStuffVerifierImpl impl);
}

File: stuff_policies/src/main/java/android/montreal/devfest/stuff/policies/PoliciesModule.kt
package android.montreal.devfest.stuff.policies

import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
internal abstract class PoliciesModule {
    @Binds
    internal abstract fun randomStuffVerifier(impl: RandomStuffVerifierImpl): RandomStuffVerifier

    @Module
    companion object {
        @Provides
        @JvmStatic
        internal fun randomStuffVerifierImpl(): RandomStuffVerifierImpl {
            return RandomStuffVerifierImpl()
        }
    }
}
